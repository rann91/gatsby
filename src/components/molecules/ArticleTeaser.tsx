/* @jsx jsx */
import { Link } from 'gatsby'
import { Fragment } from 'react'
import { Heading, jsx, Text } from 'theme-ui'
import { Strapi_Article } from '../../typings/graphql'

const ArticleTeaser = ({ title, slug, createdAt, summary }: Strapi_Article) => (
  <Fragment>
    {title && (
      <Heading
        data-testid="article-teaser-heading"
        as="h3"
        variant="h3"
        sx={{ fontSize: [3, null, null, 4] }}>
        <Link to={`/blog/${slug}`} sx={{ variant: 'text.headingLink' }}>
          {title}
        </Link>
      </Heading>
    )}
    <Text
      data-testid="article-teaser-label"
      as="time"
      variant="label"
      sx={{
        '* + &': {
          paddingTop: 3
        }
      }}>
      {new Date(createdAt).toLocaleDateString('en-US', {
        month: 'short',
        day: 'numeric',
        year: 'numeric'
      })}
    </Text>
    {summary && (
      <Text
        data-testid="article-teaser-summary"
        variant="p"
        sx={{
          '* + &': {
            paddingTop: 3
          }
        }}>
        {summary}
      </Text>
    )}
  </Fragment>
)

export default ArticleTeaser
