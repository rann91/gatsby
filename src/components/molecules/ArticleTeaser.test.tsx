import { render } from '@testing-library/react'
import React from 'react'
import data from '../../json/article.json'
import ArticleTeaser from './ArticleTeaser'

describe('ArticleTeaser', () => {
  it('renders correctly', () => {
    const { container } = render(<ArticleTeaser {...data} />)

    expect(container).toMatchSnapshot()
  })
})
