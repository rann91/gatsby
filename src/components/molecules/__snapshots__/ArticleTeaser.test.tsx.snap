// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTeaser renders correctly 1`] = `
<div>
  <h3
    class="css-9ygwq5-ArticleTeaser"
    data-testid="article-teaser-heading"
  >
    <a
      class="css-7aa45z-ArticleTeaser"
      href="/blog/10-quick-tips-about-blogging"
    >
      10 Quick Tips About Blogging
    </a>
  </h3>
  <time
    class="css-1muxnm4-ArticleTeaser"
    data-testid="article-teaser-label"
  >
    Jul 25, 2020
  </time>
</div>
`;
